cmd/agent/subcommands/hostname/command.go

```diff
@@ -47,16 +52,56 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			)
 		},
 	}
+	getHostnameCommand.Flags().BoolVarP(&cliParams.forceLocal, "local", "l", false, "Force computing the hostname in the command line instead of the agent process")
+
 	cliParams.logLevelDefaultOff.Register(getHostnameCommand)
 	return []*cobra.Command{getHostnameCommand}
 }
 
-func getHostname(_ log.Component, _ config.Component, _ *cliParams) error {
-	hname, err := hostname.Get(context.TODO())
+func printHostname(_ log.Component, config config.Component, params *cliParams) error {
+	hname, err := getHostname(config, params)
+
 	if err != nil {
 		return fmt.Errorf("Error getting the hostname: %v", err)
 	}
 
 	fmt.Println(hname)
 	return nil
 }
+
+func getHostname(config config.Component, params *cliParams) (string, error) {
+	var hname string
+	var err error
+
+	if !params.forceLocal {
+		hname, err = getRemoteHostname(config)
```

Comment:
Very minor nit: the vars `hname` and `err` aren't used outside of this if block. You can change this to `hname, err :=` instead of having the variables declared above.

Reply:
Yes good catch ! I refactored the code so now this is not needed 👍